wpanusb.c

```diff
@@ -463,18 +476,26 @@ static int wpanusb_get_device_capabilities(struct ieee802154_hw *hw)
 		return ret;
 	}
 
+	buffer = kmalloc(sizeof(valid_channels), GFP_NOIO);
+	if (!buffer)
+		return -ENOMEM;
+	ret = wpanusb_control_recv(wpanusb, GET_SUPPORTED_CHANNELS, buffer,	sizeof(valid_channels));
+	valid_channels = *(uint32_t *)buffer;
+	if (ret < 0 || !valid_channels) {
```

Comment:
This should probably be verifying that `ret == sizeof(uint32_t)`

Reply:
fixed in updated commit